delocate/tools.py

```diff
@@ -753,19 +753,25 @@ def _delete_rpaths(
     filename : str
         filename of library
     rpaths : Iterable[str]
-        rpaths to delete
+        rpaths to delete, must not contain duplicates
```

Comment:
I feel this change makes `_delete_rpaths` too unreliable.

I'm also looking into fixing these functions, I've ended up inlining `_delete_rpaths` into `_remove_absolute_rpaths` and deleting the `_delete_rpaths` function.